src/oscar/apps/basket/views.py

```diff
@@ -332,6 +333,12 @@ def form_invalid(self, form):
         clean_msgs = [m.replace('* ', '') for m in msgs if m.startswith('* ')]
         messages.error(self.request, ",".join(clean_msgs))
 
+        # We explicitly serialize the post data into the session with the django's session JSONSerializer
+        # in case a different serializer is configured globally, this will create a security vulnerability
+        # https://docs.djangoproject.com/en/3.2/topics/http/sessions/#cookie-session-backend
```

Comment:
We serialize the POST data with JSONSerializer before adding it to the session.
Without this, we could expose us to a security vulnerability if the SESSION_SERIALIZER has been configured to  https://docs.djangoproject.com/en/3.2/topics/http/sessions/#cookie-session-backend